src/main/java/hudson/plugins/s3/cloudfront/callable/AbstractCloudFrontCallable.java

```diff
@@ -0,0 +1,35 @@
+package hudson.plugins.s3.cloudfront.callable;
+
+import hudson.util.Secret;
+
+import java.io.Serializable;
+
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.cloudfront.AmazonCloudFront;
+import com.amazonaws.services.cloudfront.AmazonCloudFrontClient;
+
+public class AbstractCloudFrontCallable implements Serializable {
+	private static final long serialVersionUID = 1L;
+
+	private final String accessKey;
+	private final Secret secretKey;
+	private final boolean useRole;
+	private AmazonCloudFront cloudFrontClient;
```

Comment:
you should mark cloudFrontClient transient since it's not serializable (and it shouldn't be serialized anyway)